src/main/webapp/app/exam/participate/exam-navigation-bar/exam-navigation-bar.component.ts

```diff
@@ -60,6 +85,18 @@ export class ExamNavigationBarComponent implements OnInit {
         this.setExerciseButtonStatus(exerciseIndex);
     }
 
+    changeExerciseById(exerciseId: number) {
```

Comment:
tbh, I struggled to understand what this and the following function are doing just by looking at their names (`changeExerciseById` vs. `updateExerciseById`). Maybe find a bit more expressive names.

Reply:
I added comments and renamed the method. It should be clearer now. Thanks.

Reply:
I think if you would call the two methods `changeExerciseByIndex` and `changeExerciseById` would be enough. No need for the very long comment. "Auxiliary method to call changeExerciseByIndex based on the unique id of the exercise" would be good IMO.